### PR TITLE
Update release process to match new trade statistics delivery

### PR DESCRIPTION
**Pull request by ripcurlx**
Also did some minor release process cleanup that was overdue

### CODE REVIEW

**Review by ghubstan — CHANGES_REQUESTED** [review]
It's not due to this commit, but a link in release-process.md is broken.
Under [Build release](https://github.com/bisq-network/bisq/blob/a95630754bb0b4aaaf3390994883ca1bfef0a308/docs/release-process.md#build-release) -> 
`Make sure all version numbers are updated (update version variables and run replace_version_number.sh).`, the link to `replace_version_number.sh` is broken.

**Review by ghubstan — CHANGES_REQUESTED** [review]
Under [Sign release on macOS](https://github.com/bisq-network/bisq/blob/a95630754bb0b4aaaf3390994883ca1bfef0a308/docs/release-process.md#sign-release-on-macos), the link to `finalize.sh` is also broken.

File: desktop/package/macosx/copy_dbs.sh [hunk 1]
```diff
@@ -2,12 +2,14 @@
...
 
+version="1.5.0"
+
 # Set BISQ_DIR as environment var to the path of your locally synced Bisq data directory e.g. BISQ_DIR=~/Library/Application\ Support/Bisq
 
 dbDir=$BISQ_DIR/btc_mainnet/db
 resDir=p2p/src/main/resources
 
-cp "$dbDir/TradeStatistics2Store" "$resDir/TradeStatistics2Store_BTC_MAINNET"
+cp "$dbDir/TradeStatistics3Store" "$resDir/TradeStatistics3Store_${version}_BTC_MAINNET"
```
Comment (chimp1984) [inline desktop/package/macosx/copy_dbs.sh:14]: We do not update with each version (pre releases/hotfixes) the TradeStatistics3Store....

**Review by chimp1984 — DISMISSED** [review]
utACK

File: desktop/package/macosx/copy_dbs.sh (hunk 1 above)
Comment (ripcurlx) [reply desktop/package/macosx/copy_dbs.sh:14]: I'm aware of it. I just wanted to add it to the script, so I don't have to copy it manually. I'll just not commit it if there is no intention to add a new trade stats store version to the release.

Comment (ripcurlx) [reply desktop/package/macosx/copy_dbs.sh:14]: I've added an additional comment to the script to make it more clear.

**Review by sqrrm — APPROVED** [review]

**Pull request merged**